ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/CheckDetailsPage.tsx

```diff
@@ -29,41 +30,47 @@ function sortCheckStats(a: ComplianceCheckStatusCount, b: ComplianceCheckStatusC
 
 function CheckDetails() {
     const { checkName, profileName } = useParams();
+    const [currentDatetime, setCurrentDatetime] = useState(new Date());
 
-    const complianceCoverageChecksURL = `${complianceEnhancedCoveragePath}/profiles/${profileName}/checks`;
-
-    const profileCheckByIdFn = useCallback(
+    const fetchCheckStats = useCallback(
         () => getComplianceProfileCheckStats(profileName, checkName),
         [profileName, checkName]
     );
-    const { data, loading: isLoading, error } = useRestQuery(profileCheckByIdFn);
+    const {
+        data: checkStats,
+        loading: isLoadingCheckStats,
+        error: checkStatsError,
+    } = useRestQuery(fetchCheckStats);
 
-    if (error) {
-        return (
-            <PageSection variant="light">
-                <TableErrorComponent
-                    error={error}
-                    message="An error occurred. Try refreshing again"
-                />
-            </PageSection>
-        );
-    }
+    const fetchCheckResults = useCallback(
+        () => getComplianceProfileCheckResult(profileName, checkName),
+        [checkName, profileName]
+    );
+    const {
+        data: checkResults,
+        loading: isLoadingCheckResults,
+        error: checkResultsError,
+    } = useRestQuery(fetchCheckResults);
 
-    if (!isLoading && !data) {
-        return (
-            <NotFoundMessage
-                title="404: Profile check does not exist."
-                message={`A profile check called "${checkName}" could not be found.`}
-            />
-        );
-    }
+    const tableState = getTableUIState({
+        isLoading: isLoadingCheckResults,
+        data: checkResults?.checkResults,
+        error: checkResultsError,
+        searchFilter: {},
+    });
+
+    useEffect(() => {
+        if (checkResults) {
+            setCurrentDatetime(new Date());
+        }
+    }, [checkResults]);
 
     const checkStatsLabels =
-        data?.checkStats
+        checkStats?.checkStats
```

Comment:
An observation for the mental slow cooker, not for change in this contribution.

The double naming convention slows down reading the code, to use tip in editor, or even read service file.
*  `checkStats?.checkStats` here
* `data: checkResults?.checkResults` above
* `<ScanConfigProfiles profiles={scanConfig.scanConfig.profiles} />` elsewhere

Reply:
I don't like it either. Something along the lines of `checkStatsResponse` could even be an improvement?

Reply:
Yes, that change would help.